File: easy/8. Node Depths/node_depths.go
package main

type BinaryTree struct {
	Value       int
	Left, Right *BinaryTree
}

func recurse(node *BinaryTree, depth int) int {
  if node == nil {
    return 0
  }
  return depth + recurse(node.Left, depth+1) + recurse(node.Right, depth+1)
}

func NodeDepths(root *BinaryTree) int {
	if root == nil {
		return 0
	}
	level := 0
	return recurse(root, level)
}

File: dsa/easy/Node Depths/node_depths.cpp
#include <iostream>

class BinaryTree {
public:
  int value;
  BinaryTree *left;
  BinaryTree *right;

  BinaryTree(int value) {
    this->value = value;
    left = nullptr;
    right = nullptr;
  }
};

int recurse(BinaryTree *node, int &level) {
  if (node == nullptr) {
    return 0;
  }
  int nl = level + 1;
  int l = recurse(node->left, nl);
  int r = recurse(node->right, nl);

  return l + r + level;
}

int nodeDepths(BinaryTree *root) {
  if (root == nullptr) {
    return 0;
  }
  int level= 0;
  int out = recurse(root, level);
  return out;
}

int main() { return 0; }
